TLM/TLM/UI/Localization/Translation.cs

```diff
@@ -255,6 +226,55 @@ public class Translation {
             }
         }
 
+        /// <summary>
+        /// Validates a language id, fixing some known issues, then checks if the language is currently supported by TM:PE.
+        /// If the language is not supported, defaults to the mods' default language.
+        /// </summary>
+        /// <param name="language">The language id to validate</param>
+        /// <returns>A valid language id that TM:PE supports.</returns>
+        private static string GetValidLanguageId(string language) {
+
+            // Game translation mods specify their own translation codes; we transpose them to codes supported by TM:PE.
+            switch (language) {
+                case "jaex": {
+                    language = "ja";
+                    break;
+                }
+
+                case "zh-cn": {
+                    language = "zh";
+                    break;
+                }
+
+                case "kr": {
+                    language = "ko";
+                    break;
+                }
+            }
+
+            // If we don't support the language, use the default language instead
+            if (!AvailableLanguageCodes.Contains(language)) {
+                language = DEFAULT_LANGUAGE_CODE;
+            }
+
+            return language;
+        }
+
+        /// <summary>
+        /// If we've not verified the language code yet, verify it and set 
+        /// </summary>
+        /// <returns>A valid language code</returns>
+        internal static string GetCurrentLanguage() {
+            if (!IsLanguageCodeVerified) {
+                // Try the language specified in config file or, if not present, use the current game language
+                // Then validate against available TM:PE languages or use mod default if not supported
+                GlobalConfig.Instance.LanguageCode = GetValidLanguageId(GlobalConfig.Instance.LanguageCode ?? LocaleManager.instance.language);
+                IsLanguageCodeVerified = true;
+            }
+
+            return GlobalConfig.Instance.LanguageCode;
```

Comment:
Can be null here when you set game default language using dropdown in options which later throw exception.
```cs
ArgumentNullException: Argument cannot be null.
Parameter name: key
  at System.Collections.Generic.Dictionary`2[System.String,System.Collections.Generic.Dictionary`2[System.String,System.String]].get_Item (System.String key) [0x00000] in <filename unknown>:0 
  at TrafficManager.UI.Localization.LookupTable.Get (System.String lang, System.String key) [0x00000] in <filename unknown>:0 
  at TrafficManager.UI.Localization.LookupTable.Get (System.String key) [0x00000] in <filename unknown>:0 
  at TrafficManager.State.OptionsGeneralTab.T (System.String key) [0x00000] in <filename unknown>:0 
...
```

Reply:
Good catch.

Also, if user sets TM:PE to use game language, and then changes the game language, TM:PE won't get updated due to `IsLanguageCodeVerified == true`.

Looks like I need to rethink this. Will update later today.

Reply:
Store in the config the fact that user has selected Game Language: a special language code something like "*" or something. Then verify function will see a * and get the game language every time.

Reply:
I'm not sure that will give any benefits over just doing the extra processing every time.

Without knowing when the user changes game language, TM:PE will always have to assume the worst - it will have to check what current game language is, and deal with potential modded language codes, which means it will then also need to check if we have translations for that language.

Some verification skipping could be handled if there's a way to get current game tick; if current tick != last checked tick, do some checking then update last checked tick.

Reply:
Turns out there's an event when game language changes!

```cs
LocaleManager.eventLocaleChanged += new LocaleManager.LocaleChangedHandler(handler);
```